### PR TITLE
sl-20: Sites GET and POST / endpoints

### PR DESCRIPTION
**Pull request by dimapopovych**
image

### CODE REVIEW

**Comment by dimapopovych** [comment]
I'm not sure if this endpoint should return createdAt and updatedAt or not

**Review by bohdan-degtiar — APPROVED** [review]

**Review by Alexandr-mas — APPROVED** [review]

File: backend/src/db/migrations/20230315143040_add_sites_table.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { type Knex } from 'knex';
+
+const TABLE_NAME = 'sites';
+
+const COLUMN_NAME = {
```
Comment (andanf-e) [inline backend/src/db/migrations/20230315143040_add_sites_table.ts]: B12
```suggestion
const ColumnName = {
```

File: backend/src/packages/sites/site.controller.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    this.siteService = siteService;
+
+    this.addRoute({
+      path: SitesApiPath.ROOT,
+      method: 'GET',
+      handler: () => this.findAll(),
+    });
+  }
+
```
Comment (andanf-e) [inline backend/src/packages/sites/site.controller.ts:60]: Check how to declare swagger route description

File: backend/src/packages/sites/site.entity.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  private constructor({
+    id,
+    name,
+    publishedUrl,
+  }: {
+    id: number;
+    name: string;
+    publishedUrl: string | null;
+  }) {
```
Comment (andanf-e) [inline backend/src/packages/sites/site.entity.ts:18]: Let's move it to type as it's duplicating in several places
`upd` looks like you've already declared such type

**Review by what1s1ove — CHANGES_REQUESTED** [review]
let's in this task also add `create` functionality

**Comment by dimapopovych** [comment]
> let's in this task also add `create` functionality

Is it about only entity create method or I should implement another POST endpoint?

**Comment by what1s1ove** [comment]
> > let's in this task also add `create` functionality
> 
> Is it about only entity create method or I should implement another POST endpoint?

all create functionality

**Comment by dimapopovych** [comment]
post response
response

File: backend/src/packages/sites/site.controller.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+   */
+
+  private async findAll(): Promise<ApiHandlerResponse> {
+    return {
+      status: HttpCode.OK,
+      payload: await this.siteService.findAll(),
+    };
+  }
+
+  private async create(
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.controller.ts:113]: add swagger

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/packages/sites/site.controller.ts (hunk 4 above)
Comment (andanf-e) [reply backend/src/packages/sites/site.controller.ts:113]: \* separate comment for separate route (split general one)

**Review by andanf-e — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**